cfgrammar/src/lib/yacc/reporting.rs

```diff
@@ -198,6 +198,9 @@ pub struct SimpleErrorFormatter<'a> {
 
 impl<'a> SimpleErrorFormatter<'a> {
     #[allow(clippy::result_unit_err)]
+    // We could be more efficient by calling `feed(src[0..max_span_offset])`
+    // but we don't have the max span offset here, it may be worth considering
+    // now though because this is pub.
     pub fn new(src: &'a str, path: &'a path::Path) -> Result<Self, ()> {
```

Comment:
I considering doing that in `format_{warnings,errors}(&self, ...` using interior mutability, generally though these are reliant on the shared reference, because of the `move` in the `impl ErrorMap` of `SimpleReport`